packages/block-editor/src/components/block-alignment-control/use-available-alignments.js

```
@@ -8,22 +8,35 @@ import { useSelect } from '@wordpress/data';
  */
 import { useLayout } from '../block-list/layout';
 import { store as blockEditorStore } from '../../store';
+import { getLayoutType } from '../../layouts';
 
 const DEFAULT_CONTROLS = [ 'left', 'center', 'right', 'wide', 'full' ];
 const WIDE_CONTROLS = [ 'wide', 'full' ];
 
 export default function useAvailableAlignments( controls = DEFAULT_CONTROLS ) {
-	const { wideControlsEnabled = false } = useSelect( ( select ) => {
-		const { getSettings } = select( blockEditorStore );
-		const settings = getSettings();
-		return {
-			wideControlsEnabled: settings.alignWide,
-		};
-	}, [] );
+	const { wideControlsEnabled = false, themeSupportsLayout } = useSelect(
+		( select ) => {
+			const { getSettings } = select( blockEditorStore );
+			const settings = getSettings();
+			return {
+				wideControlsEnabled: settings.alignWide,
+				themeSupportsLayout: settings.supportsLayout,
+			};
+		},
+		[]
+	);
 	const layout = useLayout();
-	const supportsAlignments = layout.type === 'default';
+	const layoutType = getLayoutType( layout?.type ?? 'default' );
```

Should we return the `default` layout internally in `getLayoutType` to hide the implementation details of the layout with name `default`?

---

you mean if we pass an `undefined` type?

---

Yes. I could see you have this check in many usages of `getLayoutType`.